x/mint/keeper/keeper.go

```diff
@@ -277,19 +238,98 @@ func (k Keeper) DistributeMintedCoin(ctx sdk.Context, mintedCoin sdk.Coin) error
 	distributionCoin, err := getProportions(ctx, mintedCoin, proportion)
 	if err != nil {
-		return sdk.Coin{}, err
+		return sdk.Int{}, err
 	}
 	if err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, recipientModule, sdk.NewCoins(distributionCoin)); err != nil {
-		return sdk.Coin{}, err
+		return sdk.Int{}, err
+	}
+	return distributionCoin.Amount, nil
+}
+
+// distributeDeveloperRewards distributes developer rewards from developer vesting module account
+// to the respective account receivers by weight (developerRewardsReceivers).
+// If no developer reward receivers given, funds the community pool instead.
+// Returns the total amount distributed from the developer vesting module account.
+// Updates supply offsets to reflect the amount of coins distributed. This is done so because the developer rewards distributions are
+// allocated from its own module account, not the mint module accont (TODO: next step in https://github.com/osmosis-labs/osmosis/issues/1916).
+// Returns nil on success, error otherwise.
+// With respect to input parameters, errors occur when:
+// - developerRewardsProportion is greater than 1.
+// - invalid address in developer rewards receivers.
+// - the balance of developer module account is less than totalMintedCoin * developerRewardsProportion.
+// - the balance of mint module is less than totalMintedCoin * developerRewardsProportion.
+// CONTRACT:
+// - weights in developerRewardsReceivers add up to 1.
+// - addresses in developerRewardsReceivers are valid.
+func (k Keeper) distributeDeveloperRewards(ctx sdk.Context, totalMintedCoin sdk.Coin, developerRewardsProportion sdk.Dec, developerRewardsReceivers []types.WeightedAddress) (sdk.Int, error) {
+	devRewardCoin, err := getProportions(ctx, totalMintedCoin, developerRewardsProportion)
+	if err != nil {
+		return sdk.Int{}, err
+	}
+
+	developerRewardsModuleAccountAddress := k.accountKeeper.GetModuleAddress(types.DeveloperVestingModuleAcctName)
+	developerAccountBalance := k.bankKeeper.GetBalance(ctx, developerRewardsModuleAccountAddress, totalMintedCoin.Denom)
+	if developerAccountBalance.Amount.LT(devRewardCoin.Amount) {
+		return sdk.Int{}, insufficientDevVestingBalanceError{ActualBalance: developerAccountBalance.Amount, AttemptedDistribution: devRewardCoin.Amount}
+	}
+
+	devRewardCoins := sdk.NewCoins(devRewardCoin)
+	// TODO: next step in https://github.com/osmosis-labs/osmosis/issues/1916
+	// Avoid over-allocating from the mint module address and have to later burn it here:
+	if err := k.bankKeeper.BurnCoins(ctx, types.ModuleName, devRewardCoins); err != nil {
+		return sdk.Int{}, err
+	}
+
+	// Take the current balance of the developer rewards pool and remove it from the supply offset
+	// We re-introduce the new supply at the end, in order to avoid any rounding discrepancies.
+	k.bankKeeper.AddSupplyOffset(ctx, totalMintedCoin.Denom, developerAccountBalance.Amount)
```

Comment:
Lets consider moving this line below the community pool line, and then we don't have to adjust the community pool supply offset?